src/dns/cache.rs

```diff
@@ -0,0 +1,82 @@
+use std::collections::HashMap;
+use crate::dns::DNSEntry;
+
+struct Cache {
```

Comment:
The cache has a fixed size of max_entries elements and uses a lru strategy for evicting old dns entries. It's not yet complete implemented, but the lru seems to work according to the tests